src/purejavahidapi/windows/HidDevice.java

```diff
@@ -165,30 +164,37 @@ synchronized public int setOutputReport(byte reportID, byte[] data, int length)
 			throw new IllegalArgumentException("this device supports no output reports");
 		// In Windows writeFile() to HID device data has to be preceded with the report
 		// number, regardless
-		Arrays.fill(m_OutputReportArray, (byte)0);
-		m_OutputReportArray[0] = reportID;
-		System.arraycopy(data, 0, m_OutputReportArray, 1, length);
-
+		m_OutputReportMemory.write(0, new byte[] {
+				reportID
+		}, 0, 1);
+		m_OutputReportMemory.write(1, data, 0, length);
 
 		if (!m_ForceControlOutput) {
-			Kernel32.INSTANCE.ResetEvent(m_OutputReportOverlapped.hEvent);
+			ResetEvent(m_OutputReportOverlapped.hEvent);
 			m_OutputReportOverlapped.Internal = null;
 			m_OutputReportOverlapped.InternalHigh = null;
 			m_OutputReportOverlapped.Offset = 0;
 			m_OutputReportOverlapped.OffsetHigh = 0;
 
 			// In windows always attempt to write as many bytes as there are in the longest
 			// report plus one for the report number (even if zero ie not used)
-			if (!Kernel32.INSTANCE.WriteFile(m_Handle, m_OutputReportArray, m_OutputReportLength, null, m_OutputReportOverlapped)) {
-				if (Kernel32.INSTANCE.GetLastError() != ERROR_IO_PENDING) {
+			if (!WriteFile(m_Handle, m_OutputReportMemory, m_OutputReportLength, null, m_OutputReportOverlapped)) {
+				if (GetLastError() != ERROR_IO_PENDING) {
 					// WriteFile() failed. Return error.
 					// register_error(dev, "WriteFile");
+					System.err.println("WriteFile failed with GetLastError()==" + GetLastError());
 					return -1;
 				}
 			}
 
-			if (WAIT_OBJECT_0 != Kernel32.INSTANCE.WaitForSingleObject(m_OutputReportOverlapped.hEvent, 1000)) {
-				return -2;
+			if (WAIT_OBJECT_0 != WaitForSingleObject(m_OutputReportOverlapped.hEvent, 1000)) {
+				// This could be an overkill and not needed. When there is a chance need to be tested without this block
```

Comment:
Does this need to be resolved before merging?

Reply:
I am trying to play it safe, as I don't have a way to retest. For now I wanted to check it in it was successfully tested.